Function dt not found while trying to analyse it - discovering symbols is probably not configured properly

Running rector gave me this error:

```
                                                                                                                        
 [ERROR] Could not process                                                                                              
         "/home/jbrandenburg/projects/my-site/services/drupal/public/themes/gesso/includes/gesso.drush.i
         nc" file, due to:                                                                                              
         "System error: "Function dt not found while trying to analyse it - discovering symbols is probably not         
         configured properly."                                                                                          
         Run Rector with "--debug" option and post the report here: https://github.com/rectorphp/rector/issues/new". On line: 247
```

Running with debug gave me this stack trace:

```
 [ERROR] Could not process                                                                                              
         "/home/jbrandenburg/projects/my-site/services/drupal/public/themes/gesso/includes/gesso.drush.i
         nc" file, due to:                                                                                              
         "System error: "Function dt not found while trying to analyse it - discovering symbols is probably not         
         configured properly."                                                                                          
                                                                                                                        
         Stack trace:                                                                                                   
         #0                                                                                                             
         phar://vendor/phpstan/phpstan/phpstan.phar/src/Reflection/ReflectionProvider/MemoizingReflectionProvider.php(67
         ): PHPStan\Reflection\BetterReflection\BetterReflectionProvider->getFunction()                                 
         #1                                                                                                             
         vendor/rector/rector/rules/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector.php(
         168): PHPStan\Reflection\ReflectionProvider\MemoizingReflectionProvider->getFunction()                         
         #2                                                                                                             
         vendor/rector/rector/rules/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector.php(
         155):                                                                                                          
         Rector\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromBooleanStrictReturnsRector->isNativeBooleanReturnT 
         ypeFuncCall()                                                                                                  
         #3                                                                                                             
         vendor/rector/rector/rules/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector.php(
         122):                                                                                                          
         Rector\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromBooleanStrictReturnsRector->hasOnlyBoolScalarRetur 
         nExprs()                                                                                                       
         #4 vendor/rector/rector/src/Rector/AbstractScopeAwareRector.php(27):                                           
         Rector\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromBooleanStrictReturnsRector->refactorWithScope()    
         #5 vendor/rector/rector/src/Rector/AbstractRector.php(139): Rector\Rector\AbstractScopeAwareRector->refactor() 
         #6 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(178):                          
         Rector\Rector\AbstractRector->enterNode()                                                                      
         #7 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(105):                          
         PhpParser\NodeTraverser->traverseArray()                                                                       
         #8 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(198):                          
         PhpParser\NodeTraverser->traverseNode()                                                                        
         #9 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(85):                           
         PhpParser\NodeTraverser->traverseArray()                                                                       
         #10 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(47):                              
         PhpParser\NodeTraverser->traverse()                                                                            
         #11 vendor/rector/rector/src/Application/FileProcessor.php(113):                                               
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse()                                                 
         #12 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(184):                                    
         Rector\Application\FileProcessor->processFile()                                                                
         #13 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(161):                                    
         Rector\Application\ApplicationFileProcessor->processFile()                                                     
         #14 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(138):                                    
         Rector\Application\ApplicationFileProcessor->processFiles()                                                    
         #15 vendor/rector/rector/src/Console/Command/ProcessCommand.php(149):                                          
         Rector\Application\ApplicationFileProcessor->run()                                                             
         #16 vendor/rector/rector/vendor/symfony/console/Command/Command.php(327):                                      
         Rector\Console\Command\ProcessCommand->execute()                                                               
         #17 vendor/rector/rector/vendor/symfony/console/Application.php(960):                                          
         RectorPrefix202409\Symfony\Component\Console\Command\Command->run()                                            
         #18 vendor/rector/rector/vendor/symfony/console/Application.php(333):                                          
         RectorPrefix202409\Symfony\Component\Console\Application->doRunCommand()                                       
         #19 vendor/rector/rector/src/Console/ConsoleApplication.php(53):                                               
         RectorPrefix202409\Symfony\Component\Console\Application->doRun()                                              
         #20 vendor/rector/rector/vendor/symfony/console/Application.php(216):                                          
         Rector\Console\ConsoleApplication->doRun()                                                                     
         #21 vendor/rector/rector/bin/rector.php(130): RectorPrefix202409\Symfony\Component\Console\Application->run()  
         #22 vendor/rector/rector/bin/rector(5): require_once('...')                                                    
         #23 vendor/bin/rector(119): include('...')                                                                     
         #24 {main}". On line: 247    
```

The referenced file includes several instances of calls to the `dt()` function. For example

```
function gesso_drush_help($section) {
  switch ($section) {
    case 'drush:gesso':
      return dt('This command will create a new theme based on the Gesso theme.');
  }
}
```

I'm not really sure how to configure "symbol discovery". I can't really find any documentation on that.

--- a/rules-tests/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector/Fixture/func_call_found.php.inc
+++ b/rules-tests/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector/Fixture/func_call_found.php.inc
@@ -1,0 +1,27 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromBooleanStrictReturnsRector\Fixture;
+
+class FuncCallFound
+{
+    public function run()
+    {
+        return in_array('a', ['a', 'b'], true);
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromBooleanStrictReturnsRector\Fixture;
+
+class FuncCallFound
+{
+    public function run(): bool
+    {
+        return in_array('a', ['a', 'b'], true);
+    }
+}
+
+?>

--- a/rules-tests/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector/Fixture/skip_func_call_not_found.php.inc
+++ b/rules-tests/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector/Fixture/skip_func_call_not_found.php.inc
@@ -1,0 +1,11 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\BoolReturnTypeFromBooleanStrictReturnsRector\Fixture;
+
+class SkipFuncCallNotFound
+{
+    public function run()
+    {
+        return dt('some');
+    }
+}

--- a/rules/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector.php
+++ b/rules/TypeDeclaration/Rector/ClassMethod/BoolReturnTypeFromBooleanStrictReturnsRector.php
@@ -167,7 +167,12 @@ CODE_SAMPLE
             return false;
         }
 
-        $functionReflection = $this->reflectionProvider->getFunction(new Name($functionName), null);
+        $name = new Name($functionName);
+        if (! $this->reflectionProvider->hasFunction($name, null)) {
+            return false;
+        }
+
+        $functionReflection = $this->reflectionProvider->getFunction($name, null);
         if (! $functionReflection->isBuiltin()) {
             return false;
         }

